### PR TITLE
SAT: Update catalog_fixture to use the discover output

### LINKED ISSUE #5142: SAT: Fix configured_catalog stream field fulfillment
## Enviroment
- **Airbyte version**:  0.29.0-alpha
- **Severity**: Medium

## Current Behavior
SAT reads `configured_catalog` from config file and does not fill it with python-specified fields. Though `discover` command add these fields to the catalog output.

## Expected Behavior
`configured_catalog` arg in SAT should be filled the same way, as the catalog generated by `discover`.

## Steps to Reproduce
1. Specify the `primary_key` in any stream
2. Does not specify `source_defined_primary_key` in configured catalog
3. Run `discover` command
4. Debug SAT to see the `configured_catalog` values
5. Compare the output from the steps 3 and 4

## Additional Info
This issue may be very important. 
Discover command [generates](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L76-#L79) catalog with 
```
streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]
return AirbyteCatalog(streams=streams)
```
Where the `as_airbyte_stream()` method actually fills the fields with python specified data.

So I believe that we may call discover command and get its results to then use in another tests (especially `test_read` where we need generated values, but it may be needed for other tests in future).

### PR DESCRIPTION
**Pull request by htrueman**
## What
Closes #5142

## How
Update the `catalog_fixture` to get the `discover` output and generate the `AirbyteCatalog` based on it.

## Recommended reading order
1. `conftest.py`
2. `test_core.py`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by avida — APPROVED** [review]

**Review by Zirochkaa — APPROVED** [review]

**Review by yevhenii-ldv — APPROVED** [review]

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py [hunk 1]
```diff
@@ -86,9 +86,18 @@ def configured_catalog_fixture(configured_catalog_path, catalog_schemas) -> Opti
...
-def catalog_fixture(configured_catalog: ConfiguredAirbyteCatalog) -> Optional[AirbyteCatalog]:
+def catalog_fixture(
+    configured_catalog: ConfiguredAirbyteCatalog, docker_runner: ConnectorRunner, connector_config: SecretDict
+) -> Optional[AirbyteCatalog]:
+    """Generates `AirbyteCatalog`, based on `discover` output"""
+
     if configured_catalog:
-        return AirbyteCatalog(streams=[stream.stream for stream in configured_catalog.streams])
+        output = docker_runner.call_discover(config=connector_config)
+        catalog_messages = [message for message in output if message.type == Type.CATALOG]
```
Comment (yevhenii-ldv) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py:96]: After investigating a bit of the code, I got a question, can't we add `primary_key` in the same block where we add `json_schema`? (83 line of this file)

**Review by yevhenii-ldv — CHANGES_REQUESTED** [review]
Please, look comment above

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py (hunk 1 above)
Comment (htrueman) [reply airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py:96]: That's possible, but I don't think it's the best solution.
`call_discover` output produces catalog of type `AirbyteCatalog`, so we could just filter and return it in `catalog_fixture`.
But `configured_catalog_fixture` produces `ConfiguredAirbyteCatalog`, so if we want to return here the output by the discover, then we need to extract the fields from `AirbyteCatalog` and update `ConfiguredAirbyteCatalog`. The issue there is that `call_discover` output may be changed, updated fields may be changed too, thus we may miss some important data in future.

**Review by keu — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
lgtm but not sure why we're removing the empty streams check, don't let me block if you're confident its the right thing

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py [hunk 2]
```diff
@@ -185,8 +185,6 @@ def test_read(
         if inputs.validate_schema:
             self._validate_schema(records=records, configured_catalog=configured_catalog)
 
-        self._validate_empty_streams(records=records, configured_catalog=configured_catalog, allowed_empty_streams=inputs.empty_streams)
```
[the remark this answers is not in the document]
Comment (htrueman) [reply airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:188]: oh, my bad
that's a typo
I'll fix that

**Pull request merged**